packages/example-app-base/src/components/ComponentPage/ComponentPage.tsx

```diff
@@ -49,19 +52,21 @@ export class ComponentPageBase extends React.PureComponent<IComponentPageProps>
   public render() {
     const { componentName, className, otherSections, styles, theme } = this.props;
 
+    const { showOnlyExamples } = this.context;
+
     const classNames = (this._styles = getClassNames(styles, { theme }));
 
     return (
       <div className={css(classNames.root, className)}>
         <div className={componentName}>
-          {this._getPageHeader()}
+          {!showOnlyExamples && this._getPageHeader()}
           <div className={classNames.body}>
-            {this._getOverview()}
-            {this._getBestPractices()}
+            {!showOnlyExamples && this._getOverview()}
+            {!showOnlyExamples && this._getBestPractices()}
```

Comment:
@ecraig12345 from the previous feedback you asked to somewhat refactor this. I feel like this way is the least amount of code added. The only other way was to have the check done in every renderer. Let me know if you think of a better way. I am happy to adjust.

Reply:
Could we have a different ComponentPage created somehow? Like have a "as" pattern somewhere up top with the doc.ts. Using a context here feels like a code smell of something hacky.